chart/templates/elasticsearch.yaml

```diff
@@ -0,0 +1,98 @@
+{{- if .Values.elasticsearch.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-elastic
+  labels:
+    {{ include "elastic.release_labels" . | indent 6 }}
+spec:
+  type: NodePort
+  ports:
+  - name: elastic
+    port: 9200
+  selector:
+    {{ include "elastic.release_labels" . | indent 6 }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-elastic
+  labels:
+    {{ include "elastic.release_labels" . | indent 6 }}
+spec:
+  selector:
+    matchLabels:
+      {{ include "elastic.release_labels" . | indent 8 }}
+  template:
+    metadata:
+      labels:
+        {{ include "elastic.release_labels" . | indent 10 }}
+    spec:
+      initContainers:
+      # Refer to https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html
+      # and https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-configuration-memory.html#mlockall
+      - name: increase-max-map-count
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      - name: chown-data-volume-as-elastic
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sh", "-c", "chown -R 1000:100 /usr/share/elasticsearch/data"]
+        volumeMounts:
+        - name: {{ .Release.Name }}-elastic-volume
+          mountPath: "/usr/share/elasticsearch/data"
+      #Allows ES container to disable swapping for performance boost
+      #http://man7.org/linux/man-pages/man7/capabilities.7.html
+      #https://github.com/kubernetes/kubernetes/issues/3595#issuecomment-423253874
+      securityContext:
+        fsGroup: 1000
+        capabilities:
+          add:
+            - IPC_LOCK
+            - SYS_RESOURCE
+      containers:
+      - name: elastic
+        image: docker.elastic.co/elasticsearch/elasticsearch:6.5.0
```

Comment:
Would be great to be able to define version in `values.yaml`. Add a link (https://www.docker.elastic.co/) to available images as a comment too.

Reply:
Implemented in https://github.com/wunderio/drupal-project-k8s/pull/74/commits/0afa35000af306687a4b3620d300f06cb7502fe8 .